spring-boot-autoconfigure/src/main/java/org/axonframework/springboot/autoconfig/AxonServerBusAutoConfiguration.java

```diff
@@ -0,0 +1,142 @@
+import org.axonframework.commandhandling.CommandMessage;
+import org.axonframework.commandhandling.distributed.RoutingStrategy;
+import org.axonframework.common.transaction.TransactionManager;
+import org.axonframework.eventsourcing.eventstore.EventStore;
+import org.axonframework.messaging.interceptors.CorrelationDataInterceptor;
+import org.axonframework.queryhandling.QueryBus;
+import org.axonframework.queryhandling.QueryInvocationErrorHandler;
+import org.axonframework.queryhandling.QueryMessage;
+import org.axonframework.queryhandling.QueryUpdateEmitter;
+import org.axonframework.queryhandling.SimpleQueryBus;
+import org.axonframework.serialization.Serializer;
+import org.axonframework.spring.config.AxonConfiguration;
+import org.axonframework.springboot.util.ConditionalOnMissingQualifiedBean;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.AutoConfigureBefore;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Primary;
+
+/**
+ * Configures Axon Server as implementation for the CommandBus, QueryBus and EventStore.
+ *
+ * @author Marc Gathier
+ * @since 4.0
+ */
+@Configuration
+@AutoConfigureAfter(AxonServerAutoConfiguration.class)
+@AutoConfigureBefore(AxonAutoConfiguration.class)
+@ConditionalOnClass(AxonServerConfiguration.class)
+@ConditionalOnProperty(name = "axon.axonserver.enabled", matchIfMissing = true)
+public class AxonServerBusAutoConfiguration {
+
+    @Bean
+    @Primary
+    @ConditionalOnMissingQualifiedBean(qualifier = "!localSegment", beanClass = CommandBus.class)
+    public AxonServerCommandBus axonServerCommandBus(AxonServerConnectionManager axonServerConnectionManager,
+                                                     AxonServerConfiguration axonServerConfiguration,
+                                                     @Qualifier("localSegment") CommandBus localSegment,
+                                                     @Qualifier("messageSerializer") Serializer messageSerializer,
+                                                     RoutingStrategy routingStrategy,
+                                                     CommandPriorityCalculator priorityCalculator,
+                                                     CommandLoadFactorProvider loadFactorProvider,
+                                                     TargetContextResolver<? super CommandMessage<?>> targetContextResolver) {
+        return AxonServerCommandBus.builder()
+                .axonServerConnectionManager(axonServerConnectionManager)
```

Comment:
The indentation is somewhat off here. Please try again with the style guide attached to your IDE.